app/src/main/java/com/naxtlevelofandroiddevelopment/marvelgallery/network/MarvelRepository.kt

```diff
@@ -0,0 +1,28 @@
+package com.naxtlevelofandroiddevelopment.marvelgallery.network
+
+import com.naxtlevelofandroiddevelopment.marvelgallery.model.MarvelCharacter
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.providers.Provider
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.providers.RetrofitApi
+import io.reactivex.Single
+
+interface MarvelRepository {
+
+    fun getCharacters(searchQuery: String?): Single<List<MarvelCharacter>>
+
+    class Impl(val api: MarvelApi) : MarvelRepository {
```

Comment:
Then I am using invalid way and I have no problems with testing xP Hiding object behind interfaces is actually the simplest way to allow replacement for unit tests. BDW I will add some unit tests.